File: NorthStar-Database-Analytics-Organised/R_Analytics/visualisations.R
library(readr)
library(ggplot2)
library(dplyr)

df <- read_csv("../Data/sample_northstar_data.csv")

zone_delay <- df %>%
  group_by(city_zone) %>%
  summarise(avg_delay = mean(delay_minutes))

ggplot(zone_delay, aes(x = city_zone, y = avg_delay)) +
  geom_col() +
  labs(
    title = "Average Delivery Delay by City Zone",
    x = "City Zone",
    y = "Average Delay Minutes"
  )

File: NorthStar-Database-Analytics-Organised/Python/visualisations.py
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("../Data/sample_northstar_data.csv")

zone_delay = df.groupby("city_zone")["delay_minutes"].mean().reset_index()

plt.figure(figsize=(8, 5))
plt.bar(zone_delay["city_zone"], zone_delay["delay_minutes"])
plt.xlabel("City Zone")
plt.ylabel("Average Delay Minutes")
plt.title("Average Delivery Delay by City Zone")
plt.xticks(rotation=45)
plt.tight_layout()
plt.savefig("../Visualisations/delay_by_zone.png")
plt.show()
